src/main/java/de/maxhenkel/peek/events/TooltipImageEvents.java

```diff
@@ -31,11 +32,27 @@ public static Optional<TooltipComponent> getTooltipImage(ItemStack stack, Block
             return getDispenserTooltipImage(stack);
         } else if (block instanceof HopperBlock) {
             return getHopperTooltipImage(stack);
+        } else if (block instanceof EnderChestBlock) {
+            return getEnderChestTooltipImage();
         }
 
         return null;
     }
 
+    private static Optional<TooltipComponent> getEnderChestTooltipImage() {
+
+        if (!Peek.CONFIG.peekEnderChests.get()) {
+            return null;
+        }
+        //Intended to not check the showEmptyContainers config
+        //This way ender chests are shown as empty only if they have been cached and are actually empty.
+        //When first loading, ender chest rendering is vanilla.
+        if (DataStore.ecinv.stream().allMatch(ItemStack::isEmpty)) {
```

Comment:
This wouldn't work for ender chests that are empty. Please add the null check here